packages/api/src/routes/medical/patient.ts

```diff
@@ -202,29 +205,42 @@ const resourceSchema = z.enum(resourceTypeForConsolidation).array();
 /** ---------------------------------------------------------------------------
  * GET /patient/:id/consolidated
  *
- * Returns a patient's consolidated data.
+ * Triggers a patient's consolidated data query. Results are sent through Webhook.
+ * If the query is already in progress, just return the current status.
  *
  * @param req.cxId The customer ID.
  * @param req.param.id The ID of the patient whose data is to be returned.
  * @param req.query.resources Optional comma-separated list of resources to be returned.
  * @param req.query.dateFrom Optional start date that resources will be filtered by (inclusive).
  * @param req.query.dateTo Optional end date that resources will be filtered by (inclusive).
- * @return Patient's consolidated data.
+ * @return status of querying for the Patient's consolidated data.
  */
 router.get(
   "/:id/consolidated",
   asyncHandler(async (req: Request, res: Response) => {
     const cxId = getCxIdOrFail(req);
     const patientId = getFrom("params").orFail("id", req);
     const resourcesRaw = getFrom("query").optional("resources", req);
-    const resources = resourcesRaw
-      ? resourceSchema.parse(resourcesRaw.split(",").map(r => r.trim()))
-      : undefined;
+    let resourcesUnparsed: string[];
+    try {
+      resourcesUnparsed = resourcesRaw
+        ? resourcesRaw
+            .replaceAll('"', "")
+            .replaceAll("'", "")
+            .split(",")
+            .map(r => r.trim())
+            .flatMap(filterTruthy)
+        : [];
+    } catch (err) {
+      throw new BadRequestError(
+        `Invalid resources: it must be a comma-separated list of resources`
+      );
+    }
+    const resources = resourceSchema.parse(resourcesUnparsed);
```

Comment:
Fix for [this](https://metriport.slack.com/archives/C0553DYC95L/p1692559533768899)